var/spack/repos/builtin/packages/gasnet/package.py

```diff
@@ -6,98 +6,133 @@
+    variant('mpi', default=False, description="Low-performance/portable MPI conduit")
+    variant('ibv', default=False, description="Native InfiniBand verbs conduit")
+    variant('udp', default=False, description="Portable UDP conduit, for Ethernet networks")
+
+    variant('debug', default=False, description="Enable library debugging mode")
+
+    depends_on('mpi', when='+mpi')
 
-        if '+pshm' in self.spec:
-            args.append('--enable-pshm')
+    depends_on('autoconf@2.69', type='build', when='@master:')
+    depends_on('automake@1.16:', type='build', when='@master:')
+
+    def conduits(self):
+        return [c for c in ['smp', 'mpi', 'ibv', 'udp'] if "+" + c in self.spec]
+
+    def install(self, spec, prefix):
+        if spec.satisfies('@master:'):
+            bootstrapsh = Executable("./Bootstrap")
+            bootstrapsh()
+
+        # The GASNet-EX library has a highly multi-dimensional configure space,
+        # to accomodate the varying behavioral requirements of each client runtime.
+        # The library's ABI/link compatibility is strongly dependent on these
+        # client-specific build-time settings, and that variability is deliberately NOT
+        # encoded in the variants of this package. The recommended way to build/deploy
+        # GASNet is as an EMBEDDED library within the build of the client package
+        # (eg. Berkeley UPC, UPC++, Legion, etc), some of which provide build-time
+        # selection of the GASNet library sources. This spack package provides
+        # the GASNet-EX sources, for use by appropriate client packages.
+        install_tree('.', prefix + "/src")
+
+        # Library build is provided for unit-testing purposes only (see notice above)
+        if len(self.conduits()) > 0:
+            options = ["--prefix=%s" % prefix]
+
+            if '+debug' in spec:
+                options.append("--enable-debug")
+
+            if '+mpi' in spec:
+                options.append("--enable-mpi-compat")
+            else:
+                options.append("--disable-mpi-compat")
+
+            options.append("--disable-auto-conduit-detect")
+            for c in self.conduits():
+                options.append("--enable-" + c)
+
+            configure(*options)
+            make()
+            make('install')
+
+            for c in self.conduits():
+                testdir = join_path(prefix, 'tests', c)
+                mkdirp(testdir)
+                make('-C', c + '-conduit', 'testgasnet-par')
+                install(c + "-conduit/testgasnet", testdir)
+            make('-C', c + '-conduit', 'testtools-par')
+            install(c + "-conduit/testtools", join_path(prefix, 'tests'))
+
+    @run_after('install')
+    @on_package_attributes(run_tests=True)
+    def test_install(self):
+        if '+smp' in self.spec:
+            make('-C', 'smp-conduit', 'run-tests')
+        if len(self.conduits()) > 0:
+            self.run_test(join_path(prefix, 'tests', 'testtools'),
```

Comment:
While not technically necessary thanks to a bit of behind the scenes Spack magic, I prefer to be more explicit (to avoid potentially developing bad habits :grin:) in places like this, such as:

```suggestion
            self.run_test(join_path(self.prefix.tests, 'testtools'),
```

Reply:
Fixed as suggested in 83f9bf4